src/HolographicCamera.Unity/Assets/HolographicCamera/Scripts/CalibrationDataProvider.cs

```diff
@@ -57,11 +57,23 @@ private async void SendCalibrationDataAsync()
                 if (file != null)
                 {
                     byte[] contents = (await FileIO.ReadBufferAsync(file)).ToArray();
+                    if (CalculatedCameraCalibration.TryDeserialize(contents, out CalculatedCameraCalibration calibration))
+                    {
+                        // Magic offset from Unity's underlying coordinate frame (WorldManager.GetNativeISpatialCoordinateSystemPtr()) and the head pose used for the camera.
+                        // Poses are sent in the coordinate frame space because the Unity camera position uses prediction.
+                        Matrix4x4 viewFromWorld = calibration.Extrinsics.ViewFromWorld;
+                        Vector3 position = viewFromWorld.GetColumn(3);
+                        position += new Vector3(0f, 0.08f, 0.08f);
+                        viewFromWorld.SetColumn(3, position);
```

Comment:
is there more context around why this is moved to here?

Reply:
Because CalibrationData is also now used for AzureKinect calibration (which doesn't have a hardcoded offset like HoloLens).

---
In reply to: [384168184](https://github.com/microsoft/MixedReality-SpectatorView/pull/327#discussion_r384168184) [](ancestors = 384168184)